src/core/datastore/datastore.js

```diff
@@ -45,16 +45,21 @@ export class DataStore {
     }
 
     switch (type) {
-      case DataStoreType.Network:
-        store = new NetworkStore(collection, options);
+      case DataStoreType.Network: {
+        const processor = processorFactory.getNetworkProcessor();
```

Comment:
I don't, but inversion of control is a good idea and it would also facilitate testing. Each store has a default value for its processor, in case one is not provided. Currently, tests wouldn't work if there wasn't one.

Provided we leave it this way (with the processor passed to the constructor), I kind of want to make the processor the second argument, and leave `options` as the third. This wouldn't be a breaking change since it's an internal constructor, right? From a user's standpoint, the only way they can create a data store is through callikg `Kinvey.DataStore.collection()`, right? I think if this is true, I can rearrange the parameters. It would require quite a few more changes though - like in tests and who knows where else. How do you guys feel about it?

Reply:
I have to agree with @georgiwe that it is a better idea to pass the process in the constructor then have the class create it.

You can only create a datastore with `DataStore.collection()`. Go ahead and rearrange the arguments to the constructor.

Reply:
Ok 👍

Reply:
I rearranged the constructor parameters.